### PR TITLE
Add base django app

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

We need some preliminary work to build a comfortable development environment to integrate the xAPI library in openedx.

## Proposal

- [x] add base Django application
- [x] add application packaging
- [x] add docker environment (Dockerfile + docker-compose) based on openedx's hawthorn release
- [x] add learning locker LRS services
- [x] update documentation (README)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -30,6 +30,8 @@ RUN pip install --no-cache-dir --src /usr/local/src -e /edx/app/xapi
 RUN mkdir /edx/app/edxapp/staticfiles && \
     cp /edx/app/edxapp/edx-platform/common/static/webpack-stats.json /edx/app/edxapp/staticfiles/
 
+# Install dockerize to wait for mysql before running the container command
+# (and prevent connection issues)
```
Comment (jmaupetit) [inline Dockerfile:43]: I know @sampaccoud would have been unhappy without this comment :wink:

Comment (sampaccoud) [reply Dockerfile:43]: :thinking:

File: .env [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+EMAIL_PORT=1025
+
+# -- Python --
+PYTHONUNBUFFERED=1
+
+# -- Queues --
+CELERY_ALWAYS_EAGER=1
+
+# -- Features --
+FEATURES={"AUTOMATIC_AUTH_FOR_TESTING": true, "RESTRICT_AUTOMATIC_AUTH": false}
```
Comment (sampaccoud) [inline .env:29]: Maybe add one line to explain what this unlocks?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+COPY --chown=edx:edx . /edx/app/xapi/
+
+# Install development dependencies in a virtualenv
+RUN pip install --no-cache-dir --src /usr/local/src -e /edx/app/xapi
+
+# FIXME: as mentionned in fun-platform and edx-platform bug tracker, this
+# webpack-stats.json is required both in production and development in a static
+# directory 😢
+RUN mkdir /edx/app/edxapp/staticfiles && \
+    cp /edx/app/edxapp/edx-platform/common/static/webpack-stats.json /edx/app/edxapp/staticfiles/
```
Comment (sampaccoud) [inline Dockerfile:40]: but this is already done in the image you inherit from?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+ARG UID=1000
+ARG GID=1000
+
+# Switch back to a priviledged user to perform base installation
+USER root:root
+
+# Add the non-privileged user that will run the application
+RUN groupadd --gid ${GID} edx && \
+    useradd --uid ${UID} --gid ${GID} --home /edx edx
```
Comment (sampaccoud) [inline Dockerfile]: Why do we need to do this and not in fun-platform?

If necessary, edx is not a good name for this, why not use edxapp?

File: Dockerfile (hunk 1 above)
Comment (jmaupetit) [reply Dockerfile:43]: Just kidding, because I forgot this comment, and I knew you would told me to add it.

File: Dockerfile (hunk 3 above)
Comment (jmaupetit) [reply Dockerfile:40]: Unfortunately no: this file is available in the `builder` and `development` targets, but not in `production` where you are supposed to perform a `collectstatic`.

File: Dockerfile (hunk 4 above)
Comment (jmaupetit) [reply Dockerfile]: We are running this image only for local development purpose (and not production + development). This image is not supposed to be working in OpenShift or other platforms that assign a random non-privileged user ID to run a container. Hence, when working with this image, it's more **comfortable** to rely on an existing user (when you log in the container, etc.) than a random one activated _via_ an entrypoint (that is therefore not required here).

Moreover (even if it's not required in our case), the `COPY --chown` docker statement requires an existing system user (you cannot use the entrypoint trick during the build).

Concerning the group/username, `edx:edx` seemed relevant to me, but `edxapp` is also fine.

Comment (sampaccoud) [reply Dockerfile]: ok! I think this explanation needs to go somewhere in the code!

File: Dockerfile (hunk 3 above)
Comment (sampaccoud) [reply Dockerfile:40]: But if I'm not mistaken edx originally puts this file in `/edx/app/edxapp/staticfiles/` and we moved it to `/edx/app/edxapp/edx-platform/common/static/webpack-stats.json`.
So you are doing the other way around. Why not make a copy in the fun-platform image and keep it in both places?

Comment (jmaupetit) [reply Dockerfile:40]: I'm not comfortable with this as the production image is not supposed to have "collected" static files... which will be overriden _via_ a docker volume in 90% cases. I might be wrong but I think it's best to fix horrible things in derived images than in the production one. WDYT?

File: Dockerfile (hunk 4 above)
Comment (jmaupetit) [reply Dockerfile]: > ok! I think this explanation needs to go somewhere in the code!

:thinking: Sure? 

> why not use edxapp?

Should I switch to it?

File: .env (hunk 2 above)
Comment (jmaupetit) [reply .env:29]: Done.

File: Dockerfile (hunk 4 above)
Comment (sampaccoud) [reply Dockerfile]: edxapp for user: yes it's the common openedx practice.

Comment (sampaccoud) [reply Dockerfile]: > :thinking:  Sure?
I meant in the repo... doc?

**Review by sampaccoud — APPROVED** [review]

File: Dockerfile (hunk 4 above)
Comment (jmaupetit) [reply Dockerfile]: I've added a header mentioning this in the `Dockerfile`.

**Pull request merged**